module3/homework/inner_product/arithmeticAverage.cpp

```diff
@@ -0,0 +1,26 @@
+#include "arithmeticAverage.hpp"
+#include <algorithm>
+#include <numeric>
+#include <cmath>
+#include <iostream>
+#include <functional>
+
+double ArithmeticAverage(std::vector<int> vec1, std::vector<int> vec2){
+    double members = vec1.size() + vec2.size();
+    double value = std::inner_product(vec1.begin(), vec1.end(), vec2.begin(), 0, std::plus<>(), std::plus<>())/members;
+    return value;
+}
+
+
+double Distance(std::vector<int> vec1, std::vector<int> vec2){
+
+    double distance = std::inner_product(vec1.begin(), vec1.end(), vec2.begin(), 0, 
+    [](auto const& el1, auto const& el2){
```

Comment:
I suggest you to read this: https://en.cppreference.com/w/cpp/header/functional

There is a plenty of nice functions that can help you with calculator and this exercise. Example? You can use std::plus appart from this lambda. Hint - below

![Screenshot from 2020-09-08 21-42-30](https://user-images.githubusercontent.com/62703601/92520651-404cc300-f21c-11ea-85bb-49e32a0c2e44.png)

Reply:
+1